### PR TITLE
Feature/reporting system [PROGRESS MIGRATION]

### LINKED ISSUE #51: User Reporting for Manual Moderation feature
Potential tech stack:
- RabbitMQ or similar tool for message queue
- Email automation: https://resend.com/

Maybe make some kind of "admin panel" for reviewing reports
- Needs "admin" user role which we need to define in MongoDB etc

### PR DESCRIPTION
**Pull request by Gehrkej**
### Description:
---
This PR is to get the progress thus far on the Reporting system and Admin panel onto a branch in the main repository instead of the forked repository. While none of the UI that is implemented in this PR has any effect on other services, it acts as a baseline for once our database migration is complete.

**UI additions in this PR include:**
- Admin panel with 3 sections
  -  Reports
  - Users
  - Settings
- Post Report Modal
- User Report Modal
- User Management Options modal

### Additional Notes
---
- Current there is no way to get to the admin panel via the UI due to roles not being set up so you must type `/admin` in order to reach this section.
- Reports, users, and posts within the admin panel are all rendered based on the static data files found in the data directory. This will be adjusted once we get the database migrated. 

### Screenshot(s) + Gif(s):
---
image
image
image
image
image
image
image
image
image
image
image
image
image

TalkToBeavs-ReportingModal

### CODE REVIEW

File: frontend/src/components/custom/UserOptionsModal.tsx
Comment (artkolpakov) [inline frontend/src/components/custom/UserOptionsModal.tsx:22]: Rather than being declared here, this data interface should probably be moved to the [models](https://github.com/TalkToBeavs/TalkToBeavs/tree/deploy/models) directory since the submitted reports will be stored in MongoDB. However, it's fine to leave it as is for now. We just need to remember to refactor it once we implement the actual backend logic for storing the submitted reports and sending notifications about them to the admin.

**Review by artkolpakov — APPROVED** [review]
Everything LGTM, great UI additions! The next step for our team is to start working on the actual logic for storing the reports in the database and ensuring the admin receives a notification/email each time a report is submitted.

Just to be safe, I'd recommend pinging Tom or Jeff for another review in case I've missed anything. Feel free to merge it once you get the second approval 👏

File: frontend/src/components/custom/UserOptionsModal.tsx
Comment (Gehrkej) [reply frontend/src/components/custom/UserOptionsModal.tsx:22]: Agreed that it should be moved into the models directory, I believe my thought process for the time being was just having it there for the time being until we create the models for prisma and then learning how to integrate with that since we are switching to postgresql. I will be sure to move the all data interfaces to this directory upon implementation. Thanks for the feedback 😄

**Pull request merged**
